packages/math/least_common_multiple.pony

```diff
@@ -0,0 +1,24 @@
+primitive LeastCommonMultiple
+  """
+  Get the least common multiple of x and y where both x and y >= 1.
+
+  Providing 0 or numbers that overflow the integer width will result in an
+  error.
+
+  Example usage:
+
+  ```pony
+  use "math"
+
+  actor Main
+    new create(env: Env) =>
+      try
+        let lcm = LeastCommonMultiple[U64](10, 20)?
+        env.out.print(lcm.string())
+      else
+        env.out.print("No LCM")
+      end
+  ```
+  """
+  fun apply[A: (Integer[A] val & Unsigned)](x: A, y: A): A ? =>
+    (x *? y) / GreatestCommonDivisor[A](x, y)?
```

Comment:
The multiplication can overflow even when it can return a valid result. Eg. this prints `Error!`:

```pony
let x: I32 = 2147483640
let y: I32 = 8
try
  let result = (x *? y) / GreatestCommonDivisor[I32](x, y)?
  env.out.print(result.string())
else
  env.out.print("Error!")
end
```

However, since the GCD can properly divide any of the two numbers, the order of operations can be changed to this, which will print the correct result (`2147483640`):

```pony
let x: I32 = 2147483640
let y: I32 = 8
try
  let result = (x / GreatestCommonDivisor[I32](x, y)?) *? y
  env.out.print(result.string())
else
  env.out.print("Error!")
end
```

Reply:
That's a nice change @EpicEric!

Reply:
updated.